### PR TITLE
[isoltest] Add support for test-hooks.

### PR DESCRIPTION
**Pull request by aarlt**

### CODE REVIEW

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `d9a1069d406894c68c6d64597ce754a314b5f8f8`:
```
Error: Trailing whitespace found:
test/libsolidity/SemanticTest.cpp:57:    
test/libsolidity/SemanticTest.cpp:63:    

```
Please check that your changes are working as intended.

File: test/libsolidity/SemanticTest.cpp [hunk 1]
```diff
@@ -47,6 +53,16 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
 	m_lineOffset(m_reader.lineNumber()),
 	m_enforceViaYul(enforceViaYul)
 {
+    using namespace placeholders;
+
```
Comment (stackenbotten) [inline test/libsolidity/SemanticTest.cpp:57]: Coding style error

File: test/libsolidity/SemanticTest.cpp [hunk 2]
```diff
@@ -47,6 +53,16 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
...
 	m_enforceViaYul(enforceViaYul)
 {
+    using namespace placeholders;
+    
+    auto simpleSmokeBuiltin = bind(&SemanticTest::builtinSmokeTest, this, _1);
+    addBuiltin("smoke.test0", simpleSmokeBuiltin);
+    addBuiltin("smoke.test1", simpleSmokeBuiltin);
+    addBuiltin("smoke.test2", simpleSmokeBuiltin);
+    addBuiltin("smoke_test3", simpleSmokeBuiltin);
+
```
Comment (stackenbotten) [inline test/libsolidity/SemanticTest.cpp:63]: Coding style error

**Comment by aarlt** [comment]
Replaced by #11048.

**Pull request closed without merging**